gtfu/src/main/java/gtfu/TripCollection.java

```diff
@@ -35,10 +40,21 @@ public TripCollection(String path, StopCollection stopCollection, ShapeCollectio
             String serviceID = r.get("service_id");
             String headSign = r.get("trip_headsign");
             String shapeID = r.get("shape_id");
-            Trip trip = new Trip(id, routeID, serviceID, headSign, shapeCollection.get(shapeID));
-
-            map.put(id, trip);
-            list.add(trip);
+            Shape shape = shapeCollection.get(shapeID);
+            if (shape == null) {
```

Comment:
Is this branch tested?

Reply:
Yep - tested thoroughly and successfully caught multiple missing shapes across multiple agencies